git-cliff-core/src/repo.rs

```diff
@@ -444,10 +436,57 @@ impl Repository {
 				}
 			}
 		}
-		Err(Error::RepoError(String::from("no remotes configured")))
+		Err(Error::RepoError(String::from(
+			"no remotes configured or HEAD is detached",
+		)))
 	}
 }
 
+fn find_repo_owner(url: &str) -> Result<Option<(String, String)>> {
+	let segments = url_path_segments(url).or_else(|err| {
+		if url.contains("@") && url.contains(":") && url.contains("/") {
+			ssh_path_segments(url)
+		} else {
+			Err(err)
+		}
+	})?;
+	let [repo, owner, ..] = &segments[..] else {
+		return Err(Error::RepoError(String::from(
+			"Failed to get the owner and repo",
+		)));
+	};
+	Ok(Some((owner.to_string(), repo.to_string())))
+}
+
+fn url_path_segments(url: &str) -> Result<Vec<String>> {
+	Ok(Url::parse(url.strip_suffix(".git").unwrap_or(url))?
+		.path_segments()
+		.ok_or_else(|| Error::RepoError(String::from("failed to get URL segments")))?
+		.rev()
+		.map(|x| x.to_string())
+		.collect())
+}
+
+fn ssh_path_segments(url: &str) -> Result<Vec<String>> {
+	// git@hostname:owner/repo.git
+	let [_, owner_repo, ..] = url
+		.strip_suffix(".git")
+		.unwrap_or(url)
+		.split(":")
+		.collect::<Vec<_>>()[..]
+	else {
+		return Err(Error::RepoError(String::from(
+			"Failed to get the owner and repo from ssh remote (:)",
+		)));
+	};
+	let [owner, repo] = owner_repo.split("/").collect::<Vec<_>>()[..] else {
+		return Err(Error::RepoError(String::from(
+			"Failed to get the owner and repo from ssh remote (/)",
+		)));
+	};
+	Ok(vec![repo.to_string(), owner.to_string()])
```

Comment:
This always returns a vector of 2 elements, so it would be better to return a tuple instead? (or even better, `Remote`?)